applications/visualizer/frontend/src/components/viewers/EM/EMStackTilesViewer.tsx

```diff
@@ -84,11 +41,62 @@ const newSegLayer = (dataset: Dataset, slice: number) => {
       url: getSegmentationURL(dataset, slice),
       format: new GeoJSON(),
     }),
-    style: getFeatureStyle,
     zIndex: 1,
   });
 };
 
+function neuronsStyle(feature: FeatureLike, workspace: Workspace) {
+  const neuronName = neuronFeatureName(feature);
+
+  const visibleNeurons = workspace.getVisibleNeuronsInEM();
+  const visibleNeuron = visibleNeurons.find((neuron) => neuron === neuronName);
+  const color = visibleNeuron ? workspace.visibilities[neuronName][ViewerType.EM].color : undefined;
+
+  const selectedNeuron = workspace.getViewerSelectedNeurons(ViewerType.EM).find((neuron) => neuron === neuronName);
+  if (selectedNeuron) {
+    return selectedNeuronStyle(feature, color);
+  }
+
+  if (visibleNeuron) {
+    return activeNeuronStyle(feature, color);
+  }
+
+  return null;
+}
+
+function isNeuronVisible(neuronId: string, workspace: Workspace): boolean {
+  const inVisibles = workspace.getVisibleNeuronsInEM().find((n) => n === neuronId);
+  if (inVisibles) {
+    return true;
+  }
+
+  return workspace.getViewerSelectedNeurons(ViewerType.EM).find((n) => n === neuronId) != undefined;
+}
+
+function onNeuronSelect(position: Coordinate, source: VectorSource<Feature> | undefined, workspace: Workspace) {
+  const features = source?.getFeaturesAtCoordinate(position);
+  if (!features || features.length === 0) {
+    return;
+  }
+
+  const feature = features[0];
+  const neuronName = neuronFeatureName(feature);
+
+  if (!isNeuronVisible(neuronName, workspace)) {
+    return;
+  }
+
+  const selectedNeurons = workspace.getViewerSelectedNeurons(ViewerType.EM);
```

Comment:
This part of the code would also benefit from having a `isNeuronSelected` function.